services/api/src/typeDefs.js

```diff
@@ -546,9 +537,13 @@ const typeDefs = gql`
     """
     environmentType: String
     """
-    Name of the OpenShift Project/Namespace this environment is deployed into
+    The DeploymentTarget this environment is deployed into
+    """
+    deploymentTarget: DeploymentTarget
     """
-    openshiftProjectName: String
+    Identifier of this environment in the deployment target
+    """
+    deploymentTargetId: String
```

Comment:
what is this id? the kubernetes namespace?

Reply:
on naming: `deploymentTargetId` seems more like the Id of a deploymentTarget than the namespace within it?

How does this map to the uniqueName components (environmentName.projectName.kubernetesName)

Reply:
agree, probably better call it something like `environmentTargetId`?

> How does this map to the uniqueName components (environmentName.projectName.kubernetesName)
that's a good question, my understanding is that if we query the API for an environment we don't use the `name` of environment anymore, but instead use the `uniqueName`, as the name could change and having the graphql api define 3 objects that define the environment (name, project, deploymenttarget) is quite heavy.

Reply:
yes, this is just the generic version of the `openshiftProjectName`, naming is hard, open to suggstions. It doesn't have any relation to the `uniqueName`

Reply:
my inclination would be `environmentDeploymentTargetNamespace` as in the end this would define what it is: the namespace of this environment on the DeploymentTarget, two problems though:
1. it's long (oh well?)
2. not every deploymentTarget might actually have a namespace, so this could cause issues in the future. But at the end we're gonna deploy 100% into kubernetes the next years, so maybe some hard dependencies on kubernetes might not be bad?

Reply:
hmm.... maybe we don't need this at all anymore? What if we store the environment uniqueName on the namespace instead? Is it enough that lagoon-remote can query for the lagoon environment based on uniqueName?

If that's not possible... probably we can stop relying on the `openshiftProjectName` that's used all over the other mutations and switch to `uniqueName` then this obtuse name doesn't matter as much?

Reply:
I do think that there should be one name to rule them all, but in the meantime, I think this name `deploymentTargetId` does need to be clearer (that it's not the ID of the deployment target)

Reply:
I don't have any strong opinion, but could you use `fullyQualifiedEnvironmentName`? That decouples the variable name from the deployment target implementation, and you can define what "fully qualified" means on a per-deployment-target basis.